### PR TITLE
RPC: Add universal options argument to listtransactions

### PR DESCRIPTION
**Pull request by kristapsk**
Taken from #19443. There two additional arguments `paginatebypointer` and `nextpagepointer` were added and @luke-jr proposed that they could be united with existing `skip` argument into single `options` JSON argument. This is better approach than continue adding more and more different arguments for different output options. #22775 proposes to add sorting order as additional argument, I think it's better to use the same approach there too. Having this as a separate PR will make reviews and testing simpler.

Tests for old and new way of providing `skip` argument should be added, but didn't do it here as functional tests currently don't have tests for `skip` at all. I plan to add them later in a separate PR.

### CODE REVIEW

**Review by luke-jr — APPROVED** [review]
utACK, a few minor nits

**Comment by kristapsk** [comment]
Removed unnecessary extra blank line inside `RPCHelpMan()` added by accident.

**Review by promag — COMMENTED** [review]
NACK 3e8b4dd220b2ddf8ae7f10808d529e4b09058180, why not just append `options` argument? Or in #19443 just append new arguments.

**Comment by kristapsk** [comment]
@promag That was initial way it was done, to add new positional arguments, see #14898. It was @luke-jr idea to add it instead of existing `skip` and I think it makes sense. Together `skip`, `paginatebypointer`, `nextpagepointer` and `ascending_order` can be seen as analogue to MySQL's `ORDER BY x LIMIT y,z` and they will be used most likely rarely, so not adding four additional positional arguments IMHO is better approach.

**Comment by promag** [comment]
@kristapsk I think it's mixing 2 different things: add `options` parameter VS allow shorter -cli calls. If we want to simplify/improve -cli usage then I think we should try @ryanofsky approach first before "hijacking" one argument to support the other on the server-side.

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -1412,7 +1412,11 @@ static RPCHelpMan listtransactions()
                     {"label|dummy", RPCArg::Type::STR, RPCArg::Optional::OMITTED_NAMED_ARG, "If set, should be a valid label name to return only incoming transactions\n"
                           "with the specified label, or \"*\" to disable filtering and return all transactions."},
                     {"count", RPCArg::Type::NUM, RPCArg::Default{10}, "The number of transactions to return"},
-                    {"skip", RPCArg::Type::NUM, RPCArg::Default{0}, "The number of transactions to skip"},
+                    {"options|skip", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED_NAMED_ARG, "for backward compatibility: passing in a number instead of an object will result in {\"skip\":nnn}",
```
Comment (jonatack) [inline src/wallet/rpcwallet.cpp:1415]: I'm not sure here, but previous reviews on questions like this and `doc/developer-notes.md` generally seemed to discourage mixing types / overloading:

```
- Try not to overload methods on argument type. E.g. don't make `getblock(true)`
  and `getblock("hash")` do different things.
```

Comment (luke-jr) [reply src/wallet/rpcwallet.cpp:1415]: That's about supported use, not deprecation compatibility...

Comment (meshcollider) [reply src/wallet/rpcwallet.cpp:1415]: Shouldn't the old use be deprecated with planned removal then?

**Comment by MarcoFalke** [comment]
Closing for now. Let us know when you are working on this again and if it should be reopened.

**Comment by kristapsk** [comment]
Could this be reopened? Will rebase.

**Comment by kristapsk** [comment]
Rebased

**Comment by fanquake** [comment]
Note that #19762, which is mentioned in the early commentary on this PR, has now recently been merged.

**Comment by MarcoFalke** [comment]
```
wallet/rpc/transactions.cpp:555:5: error: no matching function for call to ‘RPCHelpMan::RPCHelpMan(<brace-enclosed initializer list>)’

**Comment by kristapsk** [comment]
@MarcoFalke I cannot reproduce that compiler error. Try `make clean`?

**Comment by MarcoFalke** [comment]
Did you run `git rebase bitcoin-core/master`?

**Comment by kristapsk** [comment]
> Did you run `git rebase bitcoin-core/master`?

Ok, seeing error after rebase. Will look at this.

**Comment by kristapsk** [comment]
Rebased

**Comment by kristapsk** [comment]
Rebased

**Comment by kristapsk** [comment]
Now it builds, but backwards compatibility with passing number instead of object as a third parameter is broken, will look at it later.

**Comment by kristapsk** [comment]
> Are you still working on this? If not, maybe close for now

Will try to look at this. There was bug after rebase (see comment above), need to go through other code changes to understand properly what has changed and what causes the issue.

**Comment by glozow** [comment]
Are you still working on this?

**Comment by kristapsk** [comment]
> Are you still working on this?

Not at the moment. Feel free to close. Will open a new PR in future if will get back to this.

**Pull request closed without merging**